core/src/main/java/org/apache/iceberg/GenericManifestFile.java

```diff
@@ -399,6 +415,7 @@ public String toString() {
         .add("deleted_data_files_count", deletedFilesCount)
         .add("deleted_rows_count", deletedRowsCount)
         .add("partitions", partitions)
+        .add("key_metadata", keyMetadata == null ? "null" : "(redacted)")
```

Comment:
Given we're adding a number of encryption related PRs, most of which have very sensitive data in them (encryption keys), would it make sense to make this into a utility function, such as `EncryptionUtils.toRedactedString(byte[] value)`?

We could have a unified way of redacting, which could also reduce the null checks in the code.

It would give people the option of possibly customizing the way in which they redact keys (say, in a fork or something or via an override) that redacts in the way that allows SREs to better assist customers - for example, I can imagine that there _might_ be utility in showing the first 2 or 4 bytes or something so as to be able to check that the key is definitively not the same.

Not sure how much benefit people would have in custom redaction for stringifying, but throwing that out there as I'm curious to hear if there is any utility in not entirely redacting, from a usability / debugging standpoint.

Reply:
I would actually be more against using this approach to support redact, for 2 reasons:

1. this is only for `toString`. If you really want to see the details of the key metadata you can read through the manifest table, and that should show the information fully if you can decrypt an encrypted manifest list. I think that provides better access control. Changing `toString` feels to me like opening a backdoor.
2. if we can encrypt a manifest, we should be able to support redact configuration natively through encryption manager as a part of data masking. When we can do that, you don't need a forked version to use a custom static method for this purpose.